reconcile/openshift_namespaces.py

```diff
@@ -1,127 +1,137 @@
         if is_in_shard(shard_key):
-            namespaces.append(namespace)
+            if shard_key not in namespaces:
+                namespaces[shard_key] = namespace
+            else:
+                to_be_ignored.add(shard_key)
+    for shard_key in to_be_ignored:
+        logging.debug(
+            f"Found multiple definitions for the namespace {shard_key};"
+            " Ignoring")
+        del namespaces[shard_key]
 
-    ri = ResourceInventory()
-    settings = queries.get_app_interface_settings()
-    oc_map = OC_Map(namespaces=namespaces, integration=QONTRACT_INTEGRATION,
-                    settings=settings, internal=internal,
-                    use_jump_host=use_jump_host,
-                    thread_pool_size=thread_pool_size,
-                    init_projects=True)
-    ob.init_specs_to_fetch(
-        ri,
-        oc_map,
-        namespaces=namespaces,
-        override_managed_types=['Namespace']
-    )
-
-    desired_state = []
-    for cluster, namespace, _, _ in ri:
-        if cluster not in oc_map.clusters():
-            continue
-        desired_state.append({"cluster": cluster, "namespace": namespace})
-
-    return oc_map, desired_state
+    return list(namespaces.values())
 
 
-def check_ns_exists(spec, oc_map):
+def check_ns_exists(spec: Mapping[str, str],
+                    oc_map: OC_Map) -> Tuple[Mapping[str, str], Any]:
     cluster = spec['cluster']
     namespace = spec['namespace']
-
     try:
-        create = not oc_map.get(cluster).project_exists(namespace)
-        return spec, create
+        exists = oc_map.get(cluster).project_exists(namespace)
+        return spec, exists
     except StatusCodeError as e:
-        msg = 'cluster: {},'
-        msg += 'namespace: {},'
-        msg += 'exception: {}'
-        msg = msg.format(cluster,
-                         namespace,
-                         str(e))
+        msg = (
+            f'cluster: {cluster}, namespace: {namespace}, exception: {str(e)}'
+        )
         logging.error(msg)
 
     return spec, None
 
 
-def create_new_project(spec, oc_map):
+def manage_projects(spec: Mapping[str, str],
+                    oc_map: OC_Map, dry_run: bool) -> None:
     cluster = spec['cluster']
     namespace = spec['namespace']
+    action = spec["action"]
 
     oc = oc_map.get(cluster)
     if not oc:
         logging.log(level=oc.log_level, msg=oc.message)
         return None
-    oc.new_project(namespace)
+
+    if action == NS_ACTION_CREATE:
+        try:
+            logging.info(['create', cluster, namespace])
+            if not dry_run:
+                oc.new_project(namespace)
+        except StatusCodeError as e:
+            msg = (
+                f'cluster: {cluster}, namespace: {namespace}, '
+                f'exception: {str(e)}'
+            )
+            logging.error(msg)
+
+    elif action == NS_ACTION_DELETE:
+        logging.info(['create', cluster, namespace])
```

Comment:
Typo: `delete`.

I find a dispatching table cleaner and less error-prone here:

```python
MANAGED_ACTIONS = {
   NS_ACTION_CREATE: oc.new_project,
   NS_ACTION_DELETE: oc.delete_project
}
```
and in here
```python
    try:
       logging.info([action, cluster, namespace])
       if not dry_run:
           MANAGED_ACTIONS[action](namespace)
    except ...
```

(and I'm a big proponent of not catching exceptions but I'm not gonna block this PR on that philosophical item).